### DISCUSSION TITLE
When will  tree-sitter-scala be ready for usage in other libraries as dependencies

### DISCUSSION THREAD

**Discussion by eankit** [tree-sitter/tree-sitter-scala#163] [Q&A]
I am seeing frequent commits in tree-sitter-scala, want to use it as a dependency in some of my work, I want to understand when will this be ready for usage, also I am seeing too many commits in recent times, may I know are these commits tested or have test cases along with it?

**Comment by ckipp01** [comment]
Hey @eankit, arguably this is already ready to be used. In fact it already is in various places.

> may I know are these commits tested or have test cases along with it?

Yes you can see this the [corpus](https://github.com/tree-sitter/tree-sitter-scala/tree/master/corpus) and also [test](https://github.com/tree-sitter/tree-sitter-scala/tree/master/test).

**Comment by SethTisue** [comment]
meta-comment: could GitHub Discussions be enabled in this repo, to encourage these sorts of questions but also to keep them separate from issues?

**Reply by ckipp01** [reply]
Good call, just enabled and converted this.

**Comment by eed3si9n** [comment]
It also depends on your purpose and expectations.

https://github.com/tree-sitter/tree-sitter-scala/blob/8079aede955f95148b874819087c5741ac8b3689/script/smoke_test.sh#L5-L7

Our smoketest indicates that it parses the standard library 100%, but Scala 2 compiler or Dotty codebase is around 66%. While tree-sitter generally deals with partial failures gracefully you might try to make sure that tree-sitter actually works with the codebase that you're using, especially for more critical usage than syntax highlighting in editors.
